include/spirv-tools/libspirv.h

```diff
@@ -370,7 +370,7 @@ typedef enum {
 const char* spvTargetEnvDescription(spv_target_env env);
 
 // Creates a context object.  Returns null if env is invalid.
-spv_context spvContextCreate(spv_target_env env);
+spv_context spvContextCreate(spv_target_env env, bool permissive = false);
```

Comment:
This breaks backward compatibility.  Also, this is supposed to be a C API, so default options should not be used.

The "permissive" mode should only apply to validation, so it really should be only on the validation API.
I think the right answer is to add a new validation entry point.  Question is whether it's hardcoded to be permissive, or whether we're adding  a general way to add more kinds of options other than just allow more when unrecognized extensions are given.
Early on (before first public release) the validator had an options word with bits for various kinds of checks, e.g. ID checks vs. others.  We got rid of that because it was hard to use: only an insider really knew what checks were turned on for what bits.  But that could work again if we're careful about what options are exposed.
We'd have two:   default = check everything we can, and relaxed-when-we-see-an-unrecognized-extension.  If we can think of another coarse-grain option then this might be the right way to go.

There are two entry points right now spvValidate and spvValidateBinary.   I'd make a new entry point based on the second one.  e.g. spvValidateBinaryWithOptions